src/installer/BeatPackageCompiler/AgentCustomAction.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using Microsoft.Deployment.WindowsInstaller;
+
+namespace Elastic.PackageCompiler.Beats
+{
+    public class AgentCustomAction
+    {
+        [CustomAction]
+        public static ActionResult InstallAction(Session session)
+        {
+            try
+            {
+                // If there are no install args, we stop here
+                // // (the MSI just copied the files, there will be no agent-install)
+                if (string.IsNullOrEmpty(session["INSTALLARGS"]))
+                    return ActionResult.Success;
+
+                string install_args = session["INSTALLARGS"];
+                string install_folder = Path.Combine(session["INSTALLDIR"], session["exe_folder"]);
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+                process.StartInfo.WorkingDirectory = install_folder;
+                process.StartInfo.FileName = "elastic-agent.exe";
+                process.StartInfo.Arguments = "install -f " + install_args;
+                process.StartInfo.CreateNoWindow = true;
+                process.Start();
+                process.WaitForExit();
+
+                session.Log("Agent install return code:" + process.ExitCode);
```

Comment:
Does this error get reported to the user in an obvious way? If somebody gets a connection refused error because their Fleet URL is wrong is it easy for them to tell?

Agent needs to interact with the network in addition to the host system to enroll in Fleet, so errors will be much more frequent.